pkg/spiffe/spiffe.go

```diff
@@ -56,3 +56,16 @@ func MustGenSpiffeURI(ns, serviceAccount string) string {
 	}
 	return uri
 }
+
+// GenCustomSpiffe returns the  spiffe string that can have a custom structure
+func GenCustomSpiffe(identity string) string {
+
+	if identity == "" {
+		log.Error("spiffe identity can't be empty")
+		return ""
+	}
+
+	// replace specifial character in spiffe
+	trustDomain = strings.Replace(trustDomain, "@", ".", -1)
+	return fmt.Sprintf(Scheme+"://%s/%s", trustDomain, identity)
```

Comment:
nit: You should probably favor string concat here ... Sprintf creates a lot of garbage and is significantly slower.

```golang
// Define a new global var
var uriPrefix = Schema + "://"

// Then do this here:
return uriPrefix + trustDomain + "/" + identity + "/"
```
FYI I'm already doing this for `GetSpiffeURI` in #12044, so you can copy from there if you like.

Reply:
@nmittler thanks for the pointer!